src/app/app.module.ts

```diff
@@ -3,10 +3,20 @@ import { BrowserModule } from '@angular/platform-browser';
 
 import { AppRoutingModule } from './app-routing.module';
 import { AppComponent } from './app.component';
+import { HeaderComponent } from './shared/header/header.component';
+import { FooterComponent } from './shared/footer/footer.component';
+import { CardComponent } from './shared/card/card.component';
+import { DetailComponent } from './shared/detail/detail.component';
+import {PageNotFoundComponent} from "./user/page-not-found/page-not-found.component";
 
 @NgModule({
   declarations: [
-    AppComponent
+    AppComponent,
+    PageNotFoundComponent,
+    HeaderComponent,
```

Comment:
you can add the HeaderComponent ,FooterComponent, CardComponent in a module called sharedModule and export the module instead